main.cpp

```diff
@@ -13,7 +13,17 @@ New/This/Pointers/References conclusion
      on the lines below it, write a struct named 'HeapA' that correctly shows how to own an instance of 'A' 
          on the heap without leaking, without using smart pointers. 
  */
-
+struct A {};
+struct HeapA
+{
+    HeapA() { myA = new A; }
```

Comment:
When you have to choose where to initialize your member variables and your choices are:
Constructor Member Initializer List
Inside the Constructor's body
Try to use the Constructor Member Initializer List.
The reason is that the member variables need to be instantiated/initialized when the class is created.
When you assign to them in the constructor body, you are potentially re-initializing them, which is a waste of resources.
Take a look at the output of this program: https://godbolt.org/z/q83YM63ze
Take note of the second instance being created and destroyed due to that constructor body assignment.
- [ ] revise, use the member initializer list to initialize your member variables